src/main/java/br/com/deveficiente/calendario/controller/EventoController.java

```diff
@@ -0,0 +1,83 @@
+package br.com.deveficiente.calendario.controller;
+
+import br.com.deveficiente.calendario.controller.form.EventoForm;
+import br.com.deveficiente.calendario.model.Agenda;
+import br.com.deveficiente.calendario.model.ConvidadoEvento;
+import br.com.deveficiente.calendario.model.Evento;
+import br.com.deveficiente.calendario.model.Usuario;
+import br.com.deveficiente.calendario.repository.AgendaRepository;
+import br.com.deveficiente.calendario.repository.ConvidadoEventoRepository;
+import br.com.deveficiente.calendario.repository.EventoRepository;
+import br.com.deveficiente.calendario.repository.UsuarioRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/evento")
+public class EventoController {
+
+    private List<Usuario> usuariosConvidados;
+    @Autowired
+    private UsuarioRepository usuarioRepository;
+    @Autowired
+    private EventoRepository eventoRepository;
+    @Autowired
+    private AgendaRepository agendaRepository;
+    @Autowired
+    private ConvidadoEventoRepository convidadoEventoRepository;
+    private String message;
+
+    @PostMapping
+    public ResponseEntity cadastrar(@RequestBody @Valid EventoForm eventoForm){
+        if(validaData(eventoForm) && todosConvidadosCadastrados(eventoForm.getConvidados())){
+            Usuario usuario = (Usuario) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Para trabalhar com o usuário logado, você pode receber o objeto que implementa UserDetails como argumento do método com a annotation @AuthenticationPrincipal